src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/DetectPreviewFeatureAnalyzer.cs

```diff
@@ -626,6 +626,18 @@ private static bool SymbolIsAnnotatedAsPreview(ISymbol symbol, ConcurrentDiction
                     return true;
                 }
 
+                if (symbol.ContainingAssembly != null && symbol.ContainingAssembly.HasAttribute(previewFeatureAttribute))
+                {
+                    requiresPreviewFeaturesSymbols.GetOrAdd(symbol, true);
+                    return true;
+                }
+
+                if (symbol.ContainingModule != null && symbol.ContainingModule.HasAttribute(previewFeatureAttribute))
+                {
+                    requiresPreviewFeaturesSymbols.GetOrAdd(symbol, true);
+                    return true;
+                }
+
                 INamedTypeSymbol? parent = symbol.ContainingType;
```

Comment:
Is there any reason to use `symbol.ContainingType` instead of `symbol.ContainingSymbol` here? If it was looping through all containing symbols it should have covered the `module` and `assembly`, therefore no need to check them separately above

Reply:
> Is there any reason to use symbol.ContainingType instead of symbol.ContainingSymbol here?

Nope. They were giving the same information till today. 

> If it was looping through all containing symbols it should have covered the module and assembly (in addition to symbol.ContainingType )

Yup, you're right. I just tested locally and `ContainingSymbol` seems to containing information even from external dlls. Fixed now. Nice catch!